pkg/record/record.go

```diff
@@ -15,13 +16,29 @@ const (
 type Record struct {
 	Name     string
 	Captured time.Time
+	Item     Marshalable
 
-	Fingerprint string
-	Item        Marshalable
+	fingerprint string
 }
 
-// Filename with extension, if present
-func (r *Record) Filename() string {
+// GetFingerprint returns the fingerprint possibly using the cache
+func (r *Record) GetFingerprint() (string, error) {
+	if len(r.fingerprint) == 0 {
+		content, err := r.Item.Marshal()
+		if err != nil {
+			return "", err
+		}
+
+		h := sha256.New()
+		h.Write(content)
+		r.fingerprint = hex.EncodeToString(h.Sum(nil))
+	}
+
+	return r.fingerprint, nil
```

Comment:
Maybe this could return `content` as well, because right now we marshal each item twice right? First here and then [in the recorder](https://github.com/openshift/insights-operator/pull/603/files#diff-93cbf5b12a5398c622a4f899c81da8bdca98a02a5cf178e90f1b6b43e6b7005cR76)

Reply:
fixed